common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1233,7 +1233,12 @@ def submit_problem(self, data, override_time=False):
             # without a stack trace
             else:
                 # Translators: {msg} will be replaced with a problem's error message.
-                msg = inst.message
+                try:
+                    # only return the error value of the exception
+                    message = inst.message.split("\\n")[-2].split(": ", 1)[1]
+                    msg = cgi.escape(message)
```

Comment:
We seem to be doing a piecemeal job of escaping the message.  There's no escape performed if the IndexError happens.  And we have it on line 1239 and 1230.  We should change to do it just once, before returning.